src/manager.cpp

```diff
@@ -103,6 +108,47 @@ sdbusplus::async::task<> Manager::parseConfiguration()
     co_return;
 }
 
+// NOLINTNEXTLINE
+sdbusplus::async::task<> Manager::monitorServiceNotifications()
+{
+    lg2::debug("Monitor for sibling notifications...");
+
+    try
+    {
+        // TODO : Process the unprocessed notify requests during startup
+
+        // Start watching the NOTIFY_SERVICE_DIR
+        // Monitoring for IN_MOVED_TO only as rsync creates a temporary file in
+        // the destination and then rename to original file.
+        watch::inotify::DataWatcher notifyWatcher(
+            _ctx, IN_NONBLOCK, IN_MOVED_TO, NOTIFY_SERVICES_DIR);
+
+        while (!_ctx.stop_requested())
+        {
+            if (auto dataOperations = co_await notifyWatcher.onDataChange();
+                !dataOperations.empty())
+            {
+                // TODO: add receiver logic if need to stop notifiactions
+                // when disable sync is set to true.
+                for (const auto& [path, Op] : dataOperations)
+                {
+                    notify::NotifyService notifyService(_ctx, *_extDataIfaces,
```

Comment:
This confuses me a bit - it does a spawn in the constructor and then the class goes out of scope and is destroyed while I assume things are still running?